### PR TITLE
Improvements in VM handling

### PR DESCRIPTION
**Pull request by nbraud**
- [x] vm/guest provides a root shell on the serial console, without password: since we disable root's password, we would otherwise be unable to login, and the console is only accessible by the hypervisor.
- [x] vm/install bakes the `preseed.cfg` and `authorized_keys` files into `initrd.gz` (instead of serving over tftp):
  - we can use this, in the future, to pass secrets to the VM;
  - unlike the previous TFTP setup, it doesn't implicitely rely on the VM's network having DHCP;
  - this is simpler, as we can get rid of the install network and associated config.
- [x] vm/install does not hardcode anymore `Port 22000` and `noc_ssh_keys`.

### CODE REVIEW

**Comment by nbraud** [comment]
@equinox0815 What do you think?  There are probably more things we can remove that I missed, now that we do not rely on tftp.

Also, if/when we make an OpenSSH CA for host keys, we can easily generate and sign the host keys, then pass them to the VM through initrd, so we would get rid of the `StrictHostKeyChecking` hack.

File: ansible/roles/vm/install/tasks/main.yml [hunk 1]
```diff
@@ -1,8 +1,49 @@
...
+    - set_fact: tmpdir={{ tmpdir.stdout }}
+
+    - name: generate preseed file
+      template:
+        src: "preseed_{{ vmdistro }}-{{ vmdistcodename }}.cfg.j2"
+        dest: "{{ tmpdir }}/preseed.cfg"
+
+    - name: Extract the initramfs
+      shell: gunzip -c "{{ vm_host.installer.path }}/{{ vmdistro }}-{{ vmdistcodename }}/{{ hostvars[vmname].vm_install_cooked.arch | default('amd64') }}/initrd.gz" > "{{ tmpdir }}/initrd"
+
```
Comment (equinox0815) [inline ansible/roles/vm/install/tasks/main.yml]: Actually unpacking the initrd.gz is not necessary. See:
https://unix.stackexchange.com/questions/243657/appending-files-to-initramfs-image-reliable

The append only method has the benefit that it works even if your are not root.

Comment (nbraud) [reply ansible/roles/vm/install/tasks/main.yml]: Oh, right, gzip is appendable.

Comment (nbraud) [reply ansible/roles/vm/install/tasks/main.yml]: PS: This thing should have worked as not-root, though, as I'm wasn't modifying the original, but I agree that appending directly to (a copy of) the original file is much better.

**Comment by nbraud** [comment]
@equinox0815 Done, but I discovered a new issue: I can't use `hostvars[vmname].ansible_all_ipv[46]_addresses`, as it might not exist (the vm doesn't exist yet, or the fact isn't cached), so I tried using `hostvars[vmname].vm_network.primary.ip` and it tried to run this command:

> ssh-keygen -f ~/.ssh/known_hosts -R '[{{' "(hostvars[vm_install_host].vm_host.network.ip+/+hostvars[vm_install_host].vm_host.network.mask)" "|" "ipaddr(hostvars[vm_install_host].vm_host.network.indices[inventory_hostname])" "|" "ipaddr(address)", "}}]:22000"

Basically, it failed to run the Jinja template inside of `hostvars[vmname].vm_network.primary.ip`.  Any idea what I did wrong?

**Comment by equinox0815** [comment]
You have to use `hostvars[vmname].vm_network_cooked.primary.ip`. The reason for this is that at the moment `hostvars[vmname].vm_network` depends on values from `hostvars[vm_install_host].vm_host.network.ip` and Ansible doesn't support this double indirection. 
In order do fix this we have to evaluate `hostvars[vm_install_host].vm_network` once before we can use it. The vm-install plabook does this [here](https://github.com/realraum/noc/blob/master/ansible/vm-install.yml#L8).

This is one of the many things that i want to fix by introducing the global network config in #17.

File: ansible/hosts.ini [hunk 2]
```diff
@@ -2,6 +2,8 @@
 host_domain=realraum.at
 ansible_host={{ inventory_hostname }}.{{ host_domain }}
 ansible_user=root
+ansible_port=22000
```
Comment (equinox0815) [inline ansible/hosts.ini]: if we set this here we should also adapt the ssh snippet generation @ `localconfig` [role](https://github.com/realraum/noc/blob/master/ansible/roles/localconfig/templates/ssh/10r3.conf.j2#L16)

File: ansible/remove_known_hosts.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  hosts: localhost
+  gather_facts: no
+  tasks:
+    - command: ssh-keygen -f ~/.ssh/known_hosts -R [{{ item }}]:{{ hostvars[vmname].ansible_port }}
+      with_items:
+        - "{{ hostvars[vmname].ansible_host }}"
+        - r3-{{ vmname }}
+        - r3g-{{ vmname }}
+        - r3e-{{ vmname }}
+        - "{{ hostvars[vmname].vm_network.primary.ip }}"
```
Comment (equinox0815) [inline ansible/remove_known_hosts.yml]: as already said for now we must use: `hostvars[vmname].vm_network_cooked.primary.ip`

File: ansible/hosts.ini (hunk 2 above)
Comment (nbraud) [reply ansible/hosts.ini]: Done

**Comment by nbraud** [comment]
Thanks  :)

**Pull request merged**